x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials.tsx

```diff
@@ -35,14 +35,18 @@ import { CredentialsList } from './credentials_list';
 import { CredentialsLogic } from './credentials_logic';
 
 export const Credentials: React.FC = () => {
-  const { initializeCredentialsData, resetCredentials, showCredentialsForm } = useActions(
+  const { fetchCredentials, fetchDetails, resetCredentials, showCredentialsForm } = useActions(
     CredentialsLogic
   );
 
-  const { dataLoading, shouldShowCredentialsForm } = useValues(CredentialsLogic);
+  const { meta, dataLoading, shouldShowCredentialsForm } = useValues(CredentialsLogic);
 
   useEffect(() => {
-    initializeCredentialsData();
+    fetchCredentials();
+  }, [meta.page.current]);
+
+  useEffect(() => {
+    fetchDetails();
     return () => {
       resetCredentials();
     };
```

Comment:
I removed `initializeCredentialsData` in favor of calling `fetchCredentials` and `fetchDetails` separately. This was primarily to facilitate calling `fetchXTableData` whenever `meta.page.current` changes (a general pattern I've made to all tables).